packages/eslint-plugin-office-addins/src/rules/no-empty-load.ts

```diff
@@ -0,0 +1,84 @@
+import { TSESTree } from "@typescript-eslint/experimental-utils";
+import {
+  Reference,
+  Scope,
+  Variable,
+} from "@typescript-eslint/experimental-utils/dist/ts-eslint-scope";
+import { isGetFunction, isLoadFunction } from "../utils";
+
+export = {
+  name: "no-empty-load",
+  meta: {
+    type: <"problem" | "suggestion" | "layout">"problem",
+    messages: {
+      emptyLoad: "Calling load without any argument can slow down your add-in",
+    },
+    docs: {
+      description:
+        "Calling load without any argument can cause load unneeded data and slow down your add-in",
+      category: <
+        "Best Practices" | "Stylistic Issues" | "Variables" | "Possible Errors"
+      >"Best Practices",
+      recommended: <false | "error" | "warn">false,
+      url: "https://docs.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model?view=powerpoint-js-1.1#calling-load-without-parameters-not-recommended",
+    },
+    schema: [],
+  },
+  create: function (context: any) {
+    function isEmptyLoad(node: TSESTree.MemberExpression): boolean {
+      return (
+        isLoadFunction(node) &&
+        node.parent?.type == TSESTree.AST_NODE_TYPES.CallExpression &&
+        node.parent.arguments.length === 0
+      );
+    }
+
+    function findEmptyLoad(scope: Scope) {
+      scope.variables.forEach((variable: Variable) => {
+        let getFound: boolean = false;
+        variable.references.forEach((reference: Reference) => {
+          const node: TSESTree.Node = reference.identifier;
+
+          if (
+            reference.isWrite() &&
```

Comment:
Only need this on the outer loop since you only want to find out if the var is being written to before clearing the boolean.

Reply:
You still have the other two check you don't need on the outer loop.

Reply:
I just put the check in the inner loop, it indeed makes more sense that way.